SetFlags.cmake

```diff
@@ -94,6 +94,10 @@ macro(set_flags)
                 		OUTPUT_VARIABLE GCC_VERSION)
                 endif()
 
+		if("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "arm")
```

Comment:
According to cmake documentation, it is the CPU of the target platform, so yes, this is the right one.
http://www.cmake.org/cmake/help/v3.1/variable/CMAKE_SYSTEM_PROCESSOR.html